### DISCUSSION TITLE
Has anyone ever used gokit in real project?

### DISCUSSION THREAD

**Discussion by johnDoe2018** [go-kit/kit#1135] [Q&A]
Are there any companies who use gokit in the wild?

**Comment by karthikmuralidharan** [comment]
I remember hearing Uber using it for their microservices.

**Comment by yurishkuro** [comment]
not a lot, but yes, a few projects use it.

**Comment by groob** [comment]
We use Go-Kit in multiple projects at [github.com/kolide](https://kolide.com)

**Comment by sas1024** [comment]
Yes, of course. We are using go-kit for our microservices.

**Comment by tkanos** [comment]
In my company it's on production.

**Comment by csk6124** [comment]
We are using this library on production.

**Comment by vetcher** [comment]
In our company, all go microservices uses go-kit

**Comment by kafrax** [comment]
i will use go-kit for our company project.but meet lot problem with load balancing .first is grpc load balancing.i don't how to solve it.My project can't running. （:

**Comment by ae6rt** [comment]
We're using go-kit to support a production service mesh.

**Comment by xla** [comment]
Reddit new Ad service: https://about.sourcegraph.com/go/gophercon-2018-from-prototype-to-production-lessons-from-building-and/

**Comment by jlcrow** [comment]
Our production app IOS/Android runs on gokit microservices running in Kubernetes.

**Comment by mishudark** [comment]
At Schibsted Media Group, we run lot of microservices with Go-kit

**Comment by LasTshaMAN** [comment]
We've have been using `go-kit` at https://coins.ph/ for a year and a half in production by now,

it works fine, but:
- it has quite steep learning curve (you can easily "do it wrong")
- its a lot of boiler-plate code to read/write, that slowed us down quite a bit

in summary, I would recommend using it only when you 100% sure you'll need to implement > 1 transport (HTTP+GRPC is a classic),
otherwise, if you are most likely gonna have just a single transport, you'll be more productive without `go-kit` complexity, just don't forget to "guard" your business logic from your transport logic (so that you can change it without worrying about transport as well).

**Comment by xlvdk** [comment]
A few internal services I've written use it as well.

**Comment by selmison** [comment]
> We've have been using `go-kit` at https://coins.ph/ for a year and a half in production by now,
> 
> it works fine, but:
> 
> * it has quite steep learning curve (you can easily "do it wrong")
> * its a lot of boiler-plate code to read/write, that slowed us down quite a bit
> 
> in summary, I would recommend using it only when you 100% sure you'll need to implement > 1 transport (HTTP+GRPC is a classic),
> otherwise, if you are most likely gonna have just a single transport, you'll be more productive without `go-kit` complexity, just don't forget to "guard" your business logic from your transport logic (so that you can change it without worrying about transport as well).

How do you deal with that boilerplate? Would you recommend the goa?

**Comment by ae6rt** [comment]
We use it for an API into our service discovery framework, with the fleet composed of about 1000 members.   Works well, and if I had to do it over again, I'd still use go-kit.

**Comment by syedali25** [comment]
Hey guys Go kit is just fake it
I will suggest you , don't take it

Very dull documentation and un explained examples are there.
